unblob/handlers/archive/cpio.py

```diff
@@ -253,19 +338,145 @@ def _calculate_name_size(header) -> int:
     def _calculate_mode(header) -> int:
         return decode_int(header.c_mode, 16)
 
+    @staticmethod
+    def _calculate_dev(header) -> int:
+        return os.makedev(
+            decode_int(header.c_dev_maj, 16), decode_int(header.c_dev_min, 16)
+        )
 
-class PortableASCIIWithCRCHandler(_NewASCIICommon, _CPIOHandlerBase):
-    NAME = "cpio_portable_ascii_crc"
-    PATTERNS = [HexString("30 37 30 37 30 32 // 07 07 02")]
+    @staticmethod
+    def _calculate_rdev(header) -> int:
+        return os.makedev(
+            decode_int(header.c_rdev_maj, 16), decode_int(header.c_rdev_min, 16)
+        )
+
+
+class PortableASCIIWithCRCParser(CPIOParserBase):
```

Comment:
This is an exact duplicate of `PortableASCIIParser` above  (or I have missed the difference).

I think the original intention was that in this class there is an implementation of a CRC check against `c_chksum`, that is called from `CPIOParserBase.parse`, which has a no-op base implementation, but it was never there.
Even in the CRC check case, all the below implementation should go, and `PortableASCIIWithCRCParser` should derive from `PortableASCIIParser`, and implement only `check_crc`.

Reply:
Makes sense. Implemented your suggestion with `check_crc`.